libirods_rule_engine_plugin-audit_amqp.cpp

```diff
@@ -129,35 +130,31 @@ irods::error start(irods::default_re_ctx& _u,const std::string& _instance_name)
     pn_messenger_start(messenger);
     pn_messenger_set_blocking(messenger, false);  // do not block
     
-    json_t* obj = json_object();
-    if( !obj ) {
-        return ERROR(SYS_MALLOC_ERR, "json_object() failed");
-    }
+    json obj {};
 
     struct timeval tv;
     gettimeofday(&tv, NULL);
     unsigned long time_ms = tv.tv_sec * 1000 + tv.tv_usec / 1000;
     std::stringstream time_str; time_str << time_ms;
-    json_object_set(obj, "time_stamp", json_string(time_str.str().c_str()));
+    obj["time_stamp"] = time_str.str();
 
     char host_name[MAX_NAME_LEN];
     gethostname( host_name, MAX_NAME_LEN );
-    json_object_set(obj, "hostname", json_string(host_name));
+    obj["hostname"] = host_name;
 
     pid_t pid = getpid();
     std::stringstream pid_str; pid_str << pid;
-    json_object_set(obj, "pid", json_string(pid_str.str().c_str()));
+    obj["pid"] = pid_str.str();
 
-    json_object_set(obj, "action", json_string("START"));
+    obj["action"] = "START";
 
     std::string log_file;
     if (test_mode) {
         log_file = str(boost::format("%s/%06i.txt") % log_path_prefix % pid);
-        json_object_set(obj, "log_file", json_string(log_file.c_str()));
+        obj["log_file"] = log_file;
     }
 
-    char* tmp_buf = json_dumps( obj, JSON_INDENT( 0 ) );
-    std::string msg_str = std::string("__BEGIN_JSON__") + std::string(tmp_buf) + std::string("__END_JSON__");
+    std::string msg_str = std::string("__BEGIN_JSON__") + obj.dump() + std::string("__END_JSON__");
```

Comment:
wrapping the string literals in `std::string` is not needed.

a better way to do this would be to use libfmt or `std::string::operator+=`
```c++
const auto msg_str = fmt::format(..., obj.dump());
//
// - OR -
//
std::string msg_str = "__BEGIN_JSON__";
msg_str += obj.dump();
msg_str += "__END_JSON__";
```